### PR TITLE
Make a basic User model (including secure passwords)

### PR DESCRIPTION
**Pull request by NstTShine**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :name,  presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
```
Comment (thangtx88) [inline app/models/user.rb]: ident

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+                    uniqueness: { case_sensitive: false }
```
Comment (thangtx88) [inline app/models/user.rb]: ident

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
```
Comment (thangtx88) [inline app/models/user.rb]: before_save {self.email = email.downcase}

File: app/models/user.rb (hunk 1 above)
Comment (NstTShine) [inline app/models/user.rb]: fixed

File: app/models/user.rb (hunk 2 above)
Comment (NstTShine) [inline app/models/user.rb]: fixed

File: app/models/user.rb (hunk 3 above)
Comment (NstTShine) [inline app/models/user.rb]: fixed

Comment (NstTShine) [inline app/models/user.rb]: thank you!

**Pull request merged**